lfs_util.h

```diff
@@ -7,18 +7,22 @@
 #ifndef LFS_UTIL_H
 #define LFS_UTIL_H
 
+#include <sys/util_macro.h>
+
 // Users can override lfs_util.h with their own configuration by defining
 // LFS_CONFIG as a header file to include (-DLFS_CONFIG=lfs_config.h).
 //
 // If LFS_CONFIG is used, none of the default utils will be emitted and must be
 // provided by the config file. To start, I would suggest copying lfs_util.h
 // and modifying as needed.
-#ifdef LFS_CONFIG
-#define LFS_STRINGIZE(x) LFS_STRINGIZE2(x)
-#define LFS_STRINGIZE2(x) #x
-#include LFS_STRINGIZE(LFS_CONFIG)
-#else
 
+#if IS_ENABLED(CONFIG_LFS_READONLY)
+#define LFS_READONLY 1
+#endif
+
+#if IS_ENABLED(CONFIG_LFS_CFG)
+#include CONFIG_CUSTOM_LFS_CONFIG_FILE
+#else
```

Comment:
This is not in line with already existing solution for adding custom LittleFS configuration, as described in Zephyr issue here: https://github.com/zephyrproject-rtos/zephyr/issues/40559

Reply:
@de-nordic , these changes need this PR in [Zephyr](https://github.com/zephyrproject-rtos/zephyr/pull/41501/commits/a88fa6046e51bd57e20e52333bac3b89616bed31) , the proposal here is to use 
kconfig CONFIG_CUSTOM_LFS_CONFIG_FIL="littlefs_util.h" to specify custom lfs_utils.h, which is similar to implementation in [MBEDTLS,](https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/mbedtls/Kconfig) (MBEDTLS_CFG_FILE) instead of compiler flag solution in zephyrproject-rtos/zephyr#40559(https://github.com/zephyrproject-rtos/zephyr/issues/40559)

Reply:
Can this be used:
https://github.com/zephyrproject-rtos/littlefs/blob/zephyr/lfs_util.h#L10-L20

Reply:
@de-nordic with adding compiler switch in CMakelist.txt inside zephyr/modules/fs/littlefs, how can I customize lfs_utils.h in my app?  What I am looking here is to use my customized header file from my application, eg : I want to enable readonly feature. With compiler switch, I still will have to add another kconfig to pass the file name ..

Reply:
> @de-nordic with adding compiler switch in CMakelist.txt inside zephyr/modules/fs/littlefs, how can I customize lfs_utils.h in my app?

By giving value to `CONFIG_CUSTOM_LFS_CONFIG_FILE="<some header file path>"` that would be used in CMakeLists.txt to add line `-DLFS_CONFIG=(value of CONFIG_CUSTOM_LFS_CONFIG_FILE)`, when the config is defined.
I am not CMake surgeon but it should be possible (@tejlmand, can you maybe help here?)

> What I am looking here is to use my customized header file from my application, eg : I want to enable readonly feature. 

You can do that by command line
```
west build -b ... -d ... -- -DCONFIG_LFS_READONLY=y
```

Or by adding to project:
```
CONFIG_LFS_READONLY=y
```

You can do the same thing with `CONFIG_CUSTOM_LFS_CONFIG_FILE`, so I do not get the 

>With compiler switch, I still will have to add another kconfig to pass the file name ..

Reply:
> @de-nordic , these changes need this PR in Zephyr , the proposal here is to use
kconfig CONFIG_CUSTOM_LFS_CONFIG_FIL="littlefs_util.h" to specify custom lfs_utils.h, which is similar to implementation in MBEDTLS, (MBEDTLS_CFG_FILE) instead of compiler flag solution in zephyrproject-rtos/zephyr#40559(zephyrproject-rtos/zephyr#40559)

mbedTLS is special in the sense that it has a huge number of configuration options.
Zephyr supports majority, but simply due to the number of possible combinations in mbedTLs then it is possible to use a custom config header.
Not to mention that if you choose to use mbedTLS header, then the Kconfig config options depends on the usage of the default Zephyr config header.

> > @de-nordic with adding compiler switch in CMakelist.txt inside zephyr/modules/fs/littlefs, how can I customize lfs_utils.h in my app?
>
> By giving value to CONFIG_CUSTOM_LFS_CONFIG_FILE="<some header file path>" that would be used in CMakeLists.txt to add line -DLFS_CONFIG=(value of CONFIG_CUSTOM_LFS_CONFIG_FILE), when the config is defined.
> I am not CMake surgeon but it should be possible (@tejlmand, can you maybe help here?)

From the initial looks of it, I would say the littleFS Zephyr CMake module code should generate this header based upon the Kconfig settings.

And then Kconfig can be used to control all setting.

Based on the number of config options for little FS, this should be very simple tasks.
Will take a look.